backend/protocol_rpc/endpoint_generator.py

```
@@ -105,17 +169,20 @@ def handle_eth_methods():
+                            # Log the exception with traceback
+                            app.logger.exception(
+                                "Error forwarding single request to Hardhat"
+                            )
                             raise JSONRPCError(
-                                code=-32603,
-                                message=f"Error forwarding request to Hardhat: {str(e)}",
-                                data={"original_error": str(e)},
+                                code=-32000,  # Server error
+                                message="Network error",
+                                data=f"An internal error occurred while forwarding the request to Hardhat.",
                             )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)